fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java

```diff
@@ -141,29 +137,28 @@ public void onQRCodeDetected(Barcode barcode) {
       return;
     }
 
-    mRequestFactory.setUrl(data.server);
+    networkManager.connect(data.server);
     String channel = "/root/" + data.lao;
     disposables.add(
-        mLAORepository
-            .sendSubscribe(channel)
-            .observeOn(AndroidSchedulers.mainThread())
-            .timeout(3, TimeUnit.SECONDS)
+        networkManager
+            .getMessageSender()
+            .subscribe(channel)
+            .doFinally(this::openHome)
             .subscribe(
-                answer -> {
-                  if (answer instanceof Result) {
-                    Log.d(TAG, "got success result for subscribe to lao");
-                  } else {
-                    Log.d(
-                        TAG,
-                        "got failure result for subscribe to lao: "
-                            + ((Error) answer).getError().getDescription());
-                  }
-                  openHome();
+                () -> {
+                  Log.d(TAG, "subscribing to LAO with id " + channel);
+
+                  // Create the new LAO and add it to the LAORepository LAO lists
+                  Lao lao = new Lao(data.lao);
+                  laoRepository.getLaoById().put(channel, new LAOState(lao));
+                  laoRepository.setAllLaoSubject();
+
+                  Log.d(TAG, "got success result for subscribe to lao");
                 },
-                throwable -> {
-                  Log.d(TAG, "timed out waiting for a response for subscribe to lao", throwable);
-                  openHome(); // so that it doesn't load forever
-                }));
+                throwable ->
+                    Log.d(
+                        TAG, "timed out waiting for a response for subscribe to lao", throwable)));
```

Comment:
again, the error is incorrect. this can also happen if we're subscribing to the wrong LAO.